src/Cache/FileCache.php

```diff
@@ -90,7 +90,7 @@ protected function file(string $key): string
 					// check if the segment only contains safe characters;
 					// underscores are *not* safe to guarantee uniqueness
```

Comment:
@bastianallgeier seems like your change contradicts this comment

Reply:
Shit. That happens when you don't read the comments. @lukasbestle do you remember why they are not safe? I don't really understand the comment to be honest.

Reply:
Ah, that makes sense. I didn't understand what's meant with "special cases". Do we need to modify the Nano ID implementation to support custom alphabets? Otherwise that's indeed the nicer solution.

Reply:
What I don't really get is what the difference between `Hidehalo\Nanoid\Client::MODE_NORMAL` and `MODE_DYNAMIC` is supposed to be. The default `MODE_NORMAL` skips basically all the generator code and just uses `mt_*` functions in `Client::normalRandom()`. The `mt_*` functions are not as random as `random_bytes()` and also at this point the whole library is useless as we could just use a similar logic to `Client::normalRandom()` directly.

If we find out that we want to use `MODE_DYNAMIC`, then we can use something like this:

```php
$core      = new Hidehalo\Nanoid\Core();
$generator = new Hidehalo\Nanoid\Generator();
$alphabet  = '-0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';

$uuid = $core->random($generator, $length, $alphabet);
```

But at the moment I'm a bit confused about the Nanoid library.

Reply:
I'm wondering if we shouldn't just use `Str::random($length, 'alphaNum')` instead of the Nanoid library. Then we can avoid the extra library. And the two fewer characters (`-` and `_`) aren't going to the significantly increase the collision probability.

Reply:
I think we wanted to stick to a standard format here. I think therefore, it makes more sense to use the "official" (at least listed) PHP library for Nanoid. Whether to use normal or dynamic I also don't fully understand, especially as there eis no equivalent in the JS version. But I would stick to the documented methods, so 

```php
$client->formattedId('0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ', $length);
```

Reply:
`Str::random()` could be an option too... I mean in the beginning we wanted to go with UUIDv4 as standard. But then diverted to Nanoid because it would be less of a monster as string. But in the end, `Str::random()` could do the job as well.

Reply:
And for people that really want a specific format, `Uuid` already provides a way to customise the generator.

Reply:
I'd be fine with this 👍

Reply:
Removed the nano library and reverted Basti's change to FileCache regex.